app/components/MyComponent.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  Text,
+  View,
+  TouchableOpacity,
+  FlatList,
+  TextInput,
+} from "react-native";
+
+const MyComponent = ({ data }) => {
+  const [selectedItems, setSelectedItems] = useState([]);
+  const [dataSource, setDataSource] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    setDataSource(data);
+  }, [data]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setDataSource(data.filter((item) => item.name.includes(searchTerm)));
+    }, 1000);
+  }, [searchTerm]);
+
+  const handleSelect = (item) => {
+    setSelectedItems((currentSelectedItems) => [...currentSelectedItems, item]);
+  };
+
+  const handleClear = () => {
+    inputRef.current.clear();
+  };
+
+  return (
+    <View>
+      <TextInput
+        ref={inputRef}
+        onChangeText={setSearchTerm}
+        value={searchTerm}
+      />
+      <TouchableOpacity onPress={handleClear}>
+        <Text>Clear</Text>
+      </TouchableOpacity>
+      <FlatList
+        data={dataSource}
```

Comment:
use `filteredData` as `dataSource` state is not needed.
```suggestion
        data={filteredData}
```